### PR TITLE
feat(signup-167944525): implement the signup feature

### PR DESCRIPTION
**Pull request by j0flintking02**
#### What does this PR do?
This allows users to create an account on our platform
#### Description of Task to be completed?
- should be able to add a new user
- implement the data validation
- implements tests for the feature
#### How should this be manually tested?
- checkout to the pr and pull the changes to your local machines
- create a database
- add the connection string to the .env file
- run the DB migration run `db-migrate:dev`
- run `npm run dev`
#### What are the relevant pivotal tracker stories?
[Delivers #167944525](https://www.pivotaltracker.com/story/show/167944525)
#### Screenshots (if appropriate)
image

### CODE REVIEW

File: backend/controllers/userController.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import generatePassword from '../utils/generatePassword';
+import userService from '../services/userService';
+import generateToken from '../utils/generateToken';
+
+
+const _ = require('lodash');
+
+
+const createUser = async (req, res) => {
```
Comment (drayzii) [inline backend/controllers/userController.js]: This should be in a class

**Review by drayzii — CHANGES_REQUESTED** [review]
Let's stick to classes

File: backend/controllers/userController.js (hunk 1 above)
Comment (j0flintking02) [reply backend/controllers/userController.js]: I have changed that you could kindly review it again

**Review by drayzii — APPROVED** [review]

**Review by willywax — APPROVED** [review]
Looks good

**Review by Dkabiswa — APPROVED** [review]
LGTM

**Pull request merged**